account_payment_partner/models/account_move.py

```diff
@@ -0,0 +1,156 @@
+            else:
+                move.payment_mode_id = False
+
+    @api.depends("partner_id", "payment_mode_id")
+    def _compute_invoice_partner_bank(self):
+        for move in self:
+
+            def get_bank_id():
+                return move.commercial_partner_id.bank_ids.filtered(
+                    lambda b: b.company_id == move.company_id or not b.company_id
+                )[:1]
+
+            if move.partner_id:
+                pay_mode = move.with_context(
+                    force_company=move.company_id.id
+                ).partner_id.supplier_payment_mode_id
+                if move.type == "in_invoice":
+                    if (
+                        pay_mode
+                        and pay_mode.payment_type == "outbound"
+                        and pay_mode.payment_method_id.bank_account_required
+                        and move.commercial_partner_id.bank_ids
+                    ):
+                        move.invoice_partner_bank_id = get_bank_id()
+                    else:
+                        move.invoice_partner_bank_id = False
+                elif (
+                    pay_mode
+                    and pay_mode.payment_type == "outbound"
+                    and not pay_mode.payment_method_id.bank_account_required
+                ):
+                    self.invoice_partner_bank_id = False
+                elif not self.payment_mode_id:
+                    self.invoice_partner_bank_id = False
+            else:
+                if move.type == "in_invoice":
+                    move.invoice_partner_bank_id = False
+
+    @api.depends("type")
+    def _compute_payment_mode_filter_type_domain(self):
+        for move in self:
+            if move.type in ("out_invoice", "in_refund"):
+                move.payment_mode_filter_type_domain = "inbound"
+            elif move.type in ("in_invoice", "out_refund"):
+                move.payment_mode_filter_type_domain = "outbound"
+            else:
+                move.payment_mode_filter_type_domain = False
+
+    @api.depends("partner_id", "type")
+    def _compute_partner_bank_filter_type_domain(self):
+        for move in self:
+            if move.type in ("out_invoice", "in_refund"):
+                move.partner_bank_filter_type_domain = move.bank_partner_id
+            elif move.type in ("in_invoice", "out_refund"):
+                move.partner_bank_filter_type_domain = move.commercial_partner_id
+            else:
+                move.partner_bank_filter_type_domain = False
+
+    payment_mode_filter_type_domain = fields.Char(
+        compute="_compute_payment_mode_filter_type_domain"
+    )
+    partner_bank_filter_type_domain = fields.Many2one(
+        comodel_name="res.partner", compute="_compute_partner_bank_filter_type_domain"
+    )
+    payment_mode_id = fields.Many2one(
+        comodel_name="account.payment.mode",
+        string="Payment Mode",
+        compute=_compute_payment_mode,
```

Comment:
If you wrap the method name with quotes, like for example here `compute="_compute_payment_mode"`, you don't need to move the field definition below the compute method. This is valid in general: it's useful to limit the changes in the diff and ease the code review.